README.md

```diff
@@ -5,16 +5,56 @@
 [![Clojars Project](https://img.shields.io/clojars/v/banach.svg)](https://clojars.org/banach)
 
 
-
 This contains utilities that can be useful when working with [Manifold](https://github.com/ztellman/manifold).
 
-At the moment the library provides a way to create retry functions for deferred
-producing operations.
+## Retries
+
+Banach provides flexible tools for doing retries of functions that can fail. By
+default, it offers composable exponential backoff, maximum number of attempts,
+and hard failures (that stop future retry attempts).
+
+Minimum viable snippet (with the `banach.retry` namespace aliased as `r`):
+
+```clojure
+(r/retry f (comp (r/exponentially 2) (r/up-to 5)))
+```
+
+... which will return the result of calling a potentially deferred-returning
+function `f`, retrying on errors, up to 5 times, with an exponential delay
+(starting at 2 seconds, then waiting 2*(2 ^ n)) for each subsequent failure.
+
+Maybe you know that a particular exception means you should stop trying, e.g.
+because of an access control misconfiguration:
+
+```clojure
+(r/retry f
+  (comp
+    (r/exponentially 2)
+    (r/up-to 5)
+    (r/fatal-exception (fn [e] (= (.getMessage e) "KABOON")))))
+```
+
+(Note that the fatal exception check should probably come last in the `comp`:
+you wouldn't want to potentially back off for some number of seconds for an
+exception you know you'll never recover from anyway.)
+
+Deep down, this functionality is build on a concept called "strategies", which
+take a (potentially deferred) context and return a (potentially deferred)
+context. Because they're deferred, and because the context is a general,
```

Comment:
I'll make the antecedent unambiguous anyway; it does seem likely to trip people up.